### PR TITLE
Fix euclidean to wheel conversion bugs

### PR DESCRIPTION
**Pull request by nimazareian**
### Description
Fixed two bugs in the wheel conversion code.
Euclidean velocity is a vector of x, y, theta velocities (i.e. linear movement in the x, y direction, and the angular velocity of the robot). This is the output of the primitive executor (which gets the values from HRVO). The conversion math below takes the euclidean velocity and converts it to the velocities which the 4 wheels should be turning at.

### Testing Done
Added a set of new unit tests which pass. Tested the code on the robots as well and the robots are able to move to their destination, and rotate to the desired orientation (though they still can't do both at the same time, but that is likely related to motor ramping instead of the wheel conversion code).
Wheel angles were also updated again to be what they are in the CAD, and what I've been testing with in real-life. 

### Resolved Issues
Related to #2406

### Review Checklist
Feel free to look into the paper linked in the code and double check that the math checks out (though the inverse in the paper seems to be incorrect). Personally, I recommend that you double check that the tests make sense. They are written to be generalised and not tied to a specific set of wheel angles, as a result, they don't test that values are exact. 

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [X] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [X] **Remove all commented out code**
- [X] **Remove extra print statements**: for example, those just used for testing
- [X] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Review by LiCody — COMMENTED** [review]
LGTM, good catch on the incorrect pseudoinverse, tests make sense. 1 small nit

File: src/software/physics/euclidean_to_wheel.cpp [hunk 1]
```diff
@@ -5,52 +5,74 @@
...
+        j1, -j1, -j2, j2,
+        k1,  k1,  k2, k2;
+    // clang-format on
 }
 
-WheelSpace_t EuclideanToWheel::getWheelVelocity(EuclideanSpace_t euclidean_velocity)
+WheelSpace_t EuclideanToWheel::getWheelVelocity(EuclideanSpace_t euclidean_velocity) const
 {
-    // need to multiply the angular velocity by the wheel radius
+    // need to multiply the angular velocity by the wheel radius to
```
Comment (LiCody) [inline src/software/physics/euclidean_to_wheel.cpp]:
```suggestion
    // need to multiply the angular velocity by the robot radius to
```

Comment (nimazareian) [reply src/software/physics/euclidean_to_wheel.cpp]: Good call, updated!

**Review by LiCody — DISMISSED** [review]

**Review by JoshuaWu7 — COMMENTED** [review]
Lgtm, test case logic makes sense. Could we consider using an enum to index `calculated_wheel_speeds`?

**Comment by nimazareian** [comment]
> Could we consider using an enum to index `calculated_wheel_speeds`?

Good call, I've added a set of constants for the indices. In the future if we wanted to we can define a wrapper class around `WheelSpace_t` which has methods for getting the speeds (e.g. `wheel_velocities.frontLeft()`). But it would also be abit awkward when we're doing the matrix multiplications for the conversions.

**Review by JoshuaWu7 — APPROVED** [review]
LGTM!:+1:

**Review by itsarune — APPROVED** [review]
great work debugging this!

**Pull request merged**